README.md

```diff
@@ -56,186 +50,416 @@ npm install @dojo/has
+
+Example mixin that adds method `setState` and `readonly` `state` property:
 
 ```ts
-w(WidgetClass, properties);
-w(WidgetClass, properties, children);
+// interface for the extended API
+interface StateMixin {
+	readonly state: Readonly<any>;
+	setState(state: any): void;
+}
+
+// function that accepts a class that extends WidgetBase and returns the extended class with the StateMixin
+// behavior
+function StateMixin<T extends new(...args: any[]) => WidgetBase>(Base: T): T & (new(...args: any[]) => StateMixin) {
+	return class extends Base {
+		private _state: any;
 
-w('my-widget', properties);
-w('my-widget', properties, children);
+		public setState(state: any): void {
+			// shallow copy of the state
+			this._state = { ...this._state, ...state };
+			// invalidate the widget
+			this.invalidate();
+		}
+
+		public get state(): any {
+			return this._state;
+		}
+	};
+}
 ```
 
-The example above that uses a string for the `widgetConstructor `, is taking advantage of the [widget registry](#widget-registry) functionality.
-The widget registry allows for the lazy loading of widgets.
+Examples of Dojo 2 mixins can be seen with `ThemeableMixin` and `I18nMixin` that are described in [Classes & theming](#classes--theming) and [Internationalization](#internationalization) sections.
 
-### tsx
+### Styling & Theming
 
-In additional to the programatic functions `v` and `w`, widget-core optionally supports the use of the `jsx` syntax known as [`tsx`](https://www.typescriptlang.org/docs/handbook/jsx.html) in TypeScript.
+#### Overview
 
-To start to use `jsx` in your project the widgets need to be named with a `.tsx` extension and some configuration is required in the project's `tsconfig.json`:
+Dojo 2 widget-core provides `ThemeableMixin` to decorate a widget with theming functionality and a `@theme` decorator to specify the classes available to the widget. Both of which come from `@dojo/widget-core/mixins/Themeable`.
 
-Add the configuration options for `jsx`:
+To specify the classes for a widget, an interface needs to be imported with named exports for each class and passed to the `@theme` decorator. Importing the interface provides intellisense for the class names and passing this via the `@theme` decorator tells the `ThemeableMixin` which classes can be themed.
 
+<details><summary>Example css classes interface</summary>
+
+```typescript
+export const classNameOne: string;
+export const classNameTwo: string;
 ```
-"jsx": "react",
-"jsxFactory": "tsx",
+</details>
+
+
+**Important:** at runtime a javascript file is required to provide the processed css class names.
+
+The `ThemeableMixin` provides a method available on the instance `this.classes()` that consumes string class names and converts them into the expected class object when a widget is rendered. Classes that are passed to `this.classes()` are made available to be themed, as described [here](applying-a-theme).
```

Comment:
described [here](applying-a-theme) -> described in the [applying a theme](applying-a-theme) section